### PR TITLE
handle s3 bucket errors

### LINKED ISSUE #5: Must not attempt to set credentials if no S3 bucket is configured
Without a configured S3 bucket to store the credentials, mint should not attempt to create a password, it would be unusable anyway (and the way it works now the passwords accumulate).

### PR DESCRIPTION
**Pull request by prayerslayer**
Fixes #5

### CODE REVIEW

File: src/org/zalando/stups/mint/worker/job/run.clj [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                                                  :s3_errors 0
+                                                  :message ""}
+                                                 tokens)
+              (catch Exception e
+                (when (= 429 (:status  (ex-data e)))
+                  ; bubble up if we are rate limited
+                  (throw e))
+                (storage/update-status storage-url (:id app)
+                                                   {:has_problems true
+                                                    :s3_errors (if (= "S3Exception" (:type (ex-data e)))
```
Comment (harti2006) [inline src/org/zalando/stups/mint/worker/job/run.clj]: the else branch is missing

Comment (prayerslayer) [inline src/org/zalando/stups/mint/worker/job/run.clj]: No need for it as it will return nil.

```
user=> (if false "yo")
nil
```

Comment (harti2006) [inline src/org/zalando/stups/mint/worker/job/run.clj]: then I would prefer the (when) function ;-)

Comment (prayerslayer) [inline src/org/zalando/stups/mint/worker/job/run.clj]: But the "body" is only one form, so it doesn’t add anything, really.

```
user=> (macroexpand '(when true 0))
(if true (do 0))
```

But I’m not religious about it, we can change that.

**Pull request merged**